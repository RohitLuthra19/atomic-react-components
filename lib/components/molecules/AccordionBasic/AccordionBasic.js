// @flow
/**
 *
 * AccordionBasic
 *
 */
import React from 'react';
import styled from 'styled-components';
import type { Node } from 'react';
import styles from './AccordionBasic.style';

type Props = {
  className ? : string,
  title ? : string,
  value ? : string,
  isOpen ? : boolean,
};

const AccordionBasic = ({ className, title, value, children, isOpen }: Props): Node => (
  <div className={className}>
		<details className="accordion-item">
			<summary className="accordion-item_heading">
				{title}
			</summary>
			<p>{value}</p>
		</details>
	</div>
);

AccordionBasic.defaultProps = {
  className: '',
  title: 'Accordion Heading',
  value: 'Accordion content',
  isOpen: false,
};

export default styled(AccordionBasic)
`
  ${styles};
`;

export { AccordionBasic as AccordionBasicVanilla };
