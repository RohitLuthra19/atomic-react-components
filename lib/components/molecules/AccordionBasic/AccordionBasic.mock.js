const accordionType1 = {
  className: 'accordionType1',
  title: 'Accordion Heading 1',
  value: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  isOpen: true,
};

const accordionType2 = {
  className: 'accordionType2',
  title: 'Accordion Heading 2',
  value: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  isOpen: false,
};

const accordionType3 = {
  className: 'accordionType3',
  title: 'Accordion Heading 3',
  value: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  isOpen: false,
};

export {
  accordionType1,
  accordionType2,
  accordionType3,
};
