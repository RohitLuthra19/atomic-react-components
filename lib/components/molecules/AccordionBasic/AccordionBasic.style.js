import { css } from 'styled-components';

export default css`
	summary {
		cursor: pointer
	}
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
