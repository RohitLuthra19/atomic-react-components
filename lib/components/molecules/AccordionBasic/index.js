// @flow
export { default } from './AccordionBasic';
export { AccordionBasic } from './AccordionBasic';
