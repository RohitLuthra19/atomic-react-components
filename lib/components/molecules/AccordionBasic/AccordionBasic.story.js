import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  accordionType1,
  accordionType2,
} from './AccordionBasic.mock';

// Import Styled Component to showcase variations
import AccordionBasic from '.';

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module).addParameters({
    jest: ['AccordionBasic']
  })
  .add('AccordionBasic', () => (
    <AccordionBasic {...accordionType1}/>, <
    AccordionBasic { ...accordionType2 }
    />
  ))
  .add('AccordionBasic', () =>
    <AccordionBasic {...accordionType2} />);
