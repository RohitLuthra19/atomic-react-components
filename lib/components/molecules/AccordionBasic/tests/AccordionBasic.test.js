import { shallow } from 'enzyme';

import AccordionBasic from '../index';

describe('<AccordionBasic />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AccordionBasic />);
  });

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render a <details /> tag', () => {
    expect(wrapper.find('details').length).toEqual(1);
  });

  it('should render a <summary /> tag', () => {
    expect(wrapper.find('summary').length).toEqual(1);
  });


  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
