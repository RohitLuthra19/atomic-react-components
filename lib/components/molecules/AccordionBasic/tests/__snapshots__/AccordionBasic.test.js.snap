// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionBasic /> should render correctly 1`] = `
<Component>
  Test
</Component>
`;
